File: ruby/lib/basket_actions/basket_actions.rb
require_relative './add_to_basket_command'
require_relative './checkout_command'
require_relative './display_basket_command'

module BasketActions

	class BasketActions

		def initialize(basket, catalogue, warehouse)
			@basket = basket
			@catalogue = catalogue
			@warehouse = warehouse
		end

		def add_to_basket
			AddToBasketCommand.new(@basket, @catalogue, @warehouse)
		end

		def display_basket
			DisplayBasketCommand.new(@basket)
		end

		def checkout
			CheckoutCommand.new(@basket, @warehouse)
		end

	end
end

File: python/basket_actions/basket_actions.py
from add_to_basket_command import AddToBasketCommand
from checkout_command import CheckoutCommand
from display_basket_command import DisplayBasketCommand

class BasketActions:

    def __init__(self, basket, catalogue, warehouse):
        self.basket = basket
        self.catalogue = catalogue
        self.warehouse = warehouse

    def add_to_basket(self):
        return AddToBasketCommand(self.basket, self.catalogue, self.warehouse)

    def display_basket(self):
        return DisplayBasketCommand(self.basket)

    def checkout(self):
        return CheckoutCommand(self.basket, self.warehouse)

